doc/ansible/dev/dependent_watches.md

```diff
@@ -0,0 +1,25 @@
+# Dependent Watches
+This document describes the `watchDependentResources` option in [`watches.yaml` file](https://github.com/operator-framework/operator-sdk/blob/0136d0ec3f916364a4f005895ef12aa80ab8f2de/doc/ansible/dev/advanced_options.md#watches-file-options). It delves into what dependent resources are, why the option is required and finally gives an example.
+
+### What are dependent resources?
+In most cases, an operator creates a bunch of kubernetes resources in the cluster, that helps deploy and manage the application. For instance, the [etcd-operator](https://github.com/coreos/etcd-operator/blob/master/doc/gif/demo.gif) creates two services and a number of pods for a single `EtcdCluster` CR. In this case all the kubernetes resources created by the operator for a CR is defined as dependent resources.
+
+### Why the `watchDependentResources` option?
+When an operator requires to run a reconcile loop upon change in any of its dependent resources, a developer can set `watchDependentResources` field to true in the `watches.yaml` file. When it is enabled, reconciliation loop will be triggered and ansible code will be executed upon change in any of the dependent resources. 
```

Comment:
Explain why this is useful here IMO.

Reply:
@shawn-hurley Just added why this is useful and overview of how it is achieved, please review.